### ISSUE TITLE
Portal 2 (Steam)

### ISSUE THREAD

**Issue by TuxThePenguin0** [ptitSeb/box86-compatibility-list#48]
### Description
Seems to run perfectly (sound untested)

### Tested on
Pinebook Pro - RK3399 (Debian Bullseye chroot)

### Status
**Previous behavior:** 
None reported

**Current behavior:**
Runs without issues albeit slowly given that it's a relatively intensive game for the hardware

### Screenshots
Game running without graphical glitches (blurriness is due to game running at 720p being upscaled to 1080p)
![Game running](https://user-images.githubusercontent.com/33564505/100602733-689a1880-3358-11eb-9f3e-bc2d7dad9a89.png)

### OpenGL
OpenGL 2.0

### Additional Information
This is going to need more powerful boards to have any hope of running playably

**Comment by geerlingguy** [comment]
I was excited to give it a try on my M128-28 128-core Ampere Altra Max system, but I was hit by the same error a lot of Threadripper users seem to get: [Portal 2 crashes with too many cores](https://github.com/ValveSoftware/portal2/issues/367).

I tried disabling seven of the cores (as root, `echo 0 > /sys/devices/system/cpu/cpu{121..127}/online`), and that allowed the game to launch, but any time I tried to get past the main menus, it would freeze the game (same problem as the OP in that linked GitHub issue).

**Comment by rajdakin** [comment]
Using the new PR (once it's working), you can keep your cores up and simply configure box86 to only expose some CPUs (using `BOX86_MAXCPU=32` for example).

**Comment by geerlingguy** [comment]
@rajdakin - What hardware are you running? I'm assuming the 4070 I have would be able to use acceleration still.

**Comment by rajdakin** [comment]
I used @ptitSeb's ARM computer with a 4060. Also, the issue is the lack of multilib support in the driver (they are 64 bits thus 32 bits apps run through `box86` won't have acceleration available). This issue should hopefully be solved by the future `box32` (64-bits program emulating 32-bits programs, should be added to `box64`) when it sees the day, but that's not `box86` (also, `box32` still has no full PoC so it's far from now).

**Comment by geerlingguy** [comment]
Today I tried running Portal 2 again, had to disable a few cores again (as earlier), but Portal 2 did not crash, it ran just fine, graphically. The controls seemed a little 'sticky' sometimes. I would have to let go of the key completely to be able to use it again. Was odd but worked!

I was testing under Crossover Preview for Arm Linux, will have to test Box86/64 again soon.

**Comment by ptitSeb** [comment]
I don't think Crossover use Box64 for 32bits binary, even if wowbox64 exist.

Portal 2 from Linux Steam works with box64+box32. It migtht crash at exit tho, but you can alt-tab to steam and stop the game. The game itself runs fine.

**Comment by geerlingguy** [comment]
They mentioned in their blog post on [Arm64 support for CrossOver Preview](https://www.codeweavers.com/blog/mjohnson/2025/11/6/twist-our-arm64-heres-the-latest-crossover-preview), so I was testing that — but I want to see how it runs on Box64 as well.

They also have Cyberpunk 2077 running via Steam as well, I wonder if all the latest packages are getting close on Box64/32?

**Comment by ptitSeb** [comment]
> They mentioned in their blog post on [Arm64 support for CrossOver Preview](https://www.codeweavers.com/blog/mjohnson/2025/11/6/twist-our-arm64-heres-the-latest-crossover-preview), so I was testing that — but I want to see how it runs on Box64 as well.
> 
> They also have Cyberpunk 2077 running via Steam as well, I wonder if all the latest packages are getting close on Box64/32?

Yeah, I attended the last WineConf and we could play with a System76 Ampere machine, with box64 and with CrossOver ARM64...
